### ISSUE TITLE
[Translation] Translator::getMessages() is not part of an interface

### ISSUE THREAD

**Issue by mpdude** [symfony/symfony#14529]
The `Translator::getMessages()` method is not part of any of the interfaces `Translator` implements. But as the `Translator` is `@api`, it probably is something "official".

FWIW, inside Symfony this method is used only from tests.

Is that something that should be adressed? If not, feel free to close this issue.

**Comment by aitboudad** [comment]
It was useful before adding [getCatalogue](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Translation/TranslatorBagInterface.php#L30) until now I think we need to deprecate it.

**Comment by mpdude** [comment]
`getMessages()` can be implemented without leaking details about the (internal) organization of `MessageCatalogues` inside the `Translator`. `TranslatorBagInterface` cannot. See #14530.

**Comment by stof** [comment]
The fact that it is not part of the interface means you cannot use it when you depend on the interface (which should be the case everywhere to be compatible with composition, as used in Symfony itself)

``getMessages`` is less usable than the TranslatorBagInterface because it does not allow dealing with fallback locales properly (you don't know in which locale the messages were defined).

Btw, I think ``getMessages`` was a mistake. It was introduced in Symfony 2.6 in https://github.com/symfony/symfony/pull/9859 and then we introduced TranslatorBagInterface in 2.6 too but at a later date. ``getMessages`` should probably have been removed at such point in favor of TranslatorBagInterface
